### ISSUE TITLE
apache2 SSI and Expire blocked by opencpu

### ISSUE THREAD

**Issue by cbarbu** [opencpu/opencpu#133]
Hi, 
I'm trying to use SSI (XBitHack) and ExpiresActive directives on apache2 for html pages served with opencpu and cannot get it to work though Authentication is working fine. 

I tried with and without opencpu-cache but I always need to hit Ctrl-r to get the updated version of a page and no SSI directive get interpreted (tests with `<!--#echo var="DATE_LOCAL" -->`) 

Note that with the same configuration for `<Directory /var/www/>` pages served on `/var/www/http/` work fine. A hard link between `/var/www` to `/usr/local/lib/R/site-library/accessToCareApp/www` would also work for the SSI/ no cache part but of course loosing the benefit of opencpu.

My /etc/apache2/apache2.conf is for the opencpu location : 

``` ApacheConf
<Location /ocpu/library/accessToCareApp/www/>
        Options Indexes FollowSymLinks
        AuthType Basic
        AuthName "Practices locations"
        AuthBasicProvider  file
        AuthUserFile /usr/lib/opencpu/skaSecurity
        Require valid-user
        Options +Includes
        XBitHack on
        ExpiresActive On
        ExpiresDefault A1
        Header append Cache-Control must-revalidate
</Location>
```

I'm on Ubuntu: `Linux vagrant-ubuntu-trusty-64 3.13.0-45-generic #74-Ubuntu SMP Tue Jan 13 19:36:28 UTC 2015 x86_64 x86_64 x86_64 GNU/Linux`

Server version: Apache/2.4.7 with include, expires and headers module enabled via a2enmod. 

My ocpu/info returns: 

```
R version 3.1.2 (2014-10-31)
Platform: x86_64-pc-linux-gnu (64-bit)
locale:
 [1] LC_CTYPE=en_US.UTF-8    LC_NUMERIC=C            LC_TIME=en_US.UTF-8    
 [4] LC_COLLATE=en_US.UTF-8  LC_MONETARY=en_US.UTF-8 LC_MESSAGES=C          
 [7] LC_PAPER=C              LC_NAME=C               LC_ADDRESS=C           
[10] LC_TELEPHONE=C          LC_MEASUREMENT=C        LC_IDENTIFICATION=C    
attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     
other attached packages:
[1] opencpu_1.4.6
loaded via a namespace (and not attached):
 [1] base64enc_0.1-2    brew_1.0-6         devtools_1.6.1     evaluate_0.5.5    
 [5] formatR_1.0        grid_3.1.2         httpuv_99.999      httr_0.6.0        
 [9] jsonlite_0.9.14    knitr_1.8          lattice_0.20-29    openssl_0.2       
[13] parallel_3.1.2     RAppArmor_1.0.1.99 sendmailR_1.2-1    stringr_0.6.2     
[17] tools_3.1.2        unixtools_0.1-1   
```

**Comment by jeroenooms** [comment]
I'm not sure I understand what it is you are trying to do. Most browsers will serve a cached copy of html files unless you hit `ctrl-r`, I don't think that is related to the server config.

In the default configuration, anything under `/ocpu/` gets processed by `mod_R`. I think `XBitHack` only works when apache is serving static from a `<directory />` but I could be mistaken.

Note that you are interested to see the age of a resource, opencpu includes a `X-ocpu-time` header that displays the current local time when the resource was served. In addition if you run `opencpu-cache` there will be a `X-ocpu-cache` header that will tell if the request was served from the nginx cache.

**Comment by jeroenooms** [comment]
If you want to mess around with setting headers, this is best done by installing `opencpu-cache` and settings these headers in the nginx config, rather than in Apache.

**Comment by cbarbu** [comment]
Thanks, that was definitely nginx the missing part as it should allow to handle SSI and cache time. 

But I still don't see how the X-ocpu-time is set and how I can interact with it.

**Comment by jeroen** [comment]
Cleaning up old issues. Please open a new issue if this problem still exists in OpenCPU 2.0.
